pkg/controller/servicebindingrequest/envvars/envvars.go

```diff
@@ -0,0 +1,133 @@
+package envvars
+
+import (
+	"errors"
+	"strconv"
+	"strings"
+
+	"github.com/imdario/mergo"
+)
+
+// ErrUnsupportedType is returned when an unsupported type is encountered.
+var ErrUnsupportedType = errors.New("unsupported type")
+
+// Build returns an environment variable dictionary with an entry for each
+// leaf containing a scalar value.
+//
+// Example:
+//
+// 	src := map[string]interface{}{
+// 		"status": map[string]interface{}{
+//			"listeners": []map[string]interface{}{
+//				{
+//					"type": "secure",
+//					"addresses": []map[string]interface{}{
+//						{
+//							"host": "my-cluster-kafka-bootstrap.coffeeshop.svc",
+//							"port": "9093",
+//						},
+//					},
+//				},
+//			},
+//		},
+//	}
+//	actual, _ := Build(src)
+//
+// actual should contain the following values:
+//
+// 	"STATUS_LISTENERS_0_TYPE":             "secure",
+// 	"STATUS_LISTENERS_0_ADDRESSES_0_HOST": "my-cluster-kafka-bootstrap.coffeeshop.svc",
+// 	"STATUS_LISTENERS_0_ADDRESSES_0_PORT": "9093",
+//
+// Now, consider the following example:
+//
+//	actual, _ = Build(src, "kafka")
+//
+// actual should contain the following values instead:
+//
+// 	"KAFKA_STATUS_LISTENERS_0_TYPE":             "secure",
+// 	"KAFKA_STATUS_LISTENERS_0_ADDRESSES_0_HOST": "my-cluster-kafka-bootstrap.coffeeshop.svc",
+// 	"KAFKA_STATUS_LISTENERS_0_ADDRESSES_0_PORT": "9093",
+//
+func Build(obj interface{}, path ...string) (map[string]string, error) {
+	// perform the appropriate action depending on its type; maybe at some point
+	// reflection might be required.
+	switch val := obj.(type) {
+	case map[string]interface{}:
+		return buildMap(val, path)
+	case []map[string]interface{}:
+		return buildSliceOfMap(val, path)
+	case string:
+		return buildString(val, path), nil
+	case int:
+		return buildString(strconv.Itoa(val), path), nil
+	case int64:
+		return buildString(strconv.FormatInt(val, 10), path), nil
+	case float64:
+		return buildString(strconv.FormatFloat(val, 'f', -1, 64), path), nil
+	default:
+		return nil, ErrUnsupportedType
```

Comment:
Could there be a case with []string or []int ? And are we considering that as unsupported?

Reply:
Because it is incomplete. It should be valid to use a slice of each of the types supported by the JSON decoder.